Updating test structure

shrutivanw opened the pull request:
Updating tests to have nested describes and correct spacing. Updating readme file.
Note 1: The solution branch will be updated to have the same changes.
Note 2: all other CS Fundamentals' homework will be following this format.

shrutivanw committed:
Defining test structure for CS Fundamentals homework

Defining test structure for CS Fundamentals spec style tests. The same structure will be copied to solution's branch and followed for other homework repos.

shrutivanw committed:
Fixing spec

Fixing specs and updating readme file to share minimal requirements for homework.

shrutivanw committed:
Correcting typo

Correcting typo

kariabancroft commented on specs/string_reverse_spec.rb:
```diff
@@ -2,53 +2,70 @@
+        test_string.must_equal "olleH"
+      end
+    end
+    describe "test 2" do
```
I think you're using extra `describe` blocks here. Since this is really modeled off of a BDD framework, ideally everything that is a text block has some sort of explanation for the expected behavior. I think you could have the single `describe` block at the top-level for "basic tests" and then dive straight into the `it` blocks that have the appropriate behavior descriptions. (The "test 1" and "test 2" describe blocks are really what is concerning, so you could either add additional descriptions to these or collapse them down - whichever makes sense)

shrutivanw replied:
Sounds good! It's likely that I misunderstood Charles's feedback and am glad that you are reviewing it. Thank you.

kariabancroft commented on specs/string_reverse_spec.rb:
```diff
@@ -2,53 +2,70 @@
-    test_string.must_equal "!suoicodilaipxecitsilegarfilacrepuS .suoicocerp dnuos syawla ll'uoy ,hguone duol ti yas uoy fI .suoicorta etiuq gnihtemos si ti fo dnuos eht hguoht nevE !suoicodilaipxecitsilegarfilacrepuS ,hO"
+  # check for edge cases
+  describe "edge cases" do
+    describe "edge case 1" do
```
Same comment re: describe blocks and their descriptions

shrutivanw committed:
Removing nested describes from basic tests and edge case tests

Removing inner most describe block nested within basic tests and edge case tests.

shrutivanw commented:
Thank you, @kariabancroft PR updated. Once we get to consensus for this PR, I'll apply the same fix to the other PRs.

Hamled commented on README.md:
```diff
@@ -3,5 +3,7 @@ In this assignment, you'll design and implement one of the string manipulation f
+* Design and implement a method to reverse a string *in place*. For example, if the method is called with input parameter of "Lovelace", the method should update the input string object to have the value "ecalevoL".
+* Share and explain the time and space complexities for your solution.
+* At minimum, your implementation should pass the basic tests.
+**Note**: Do not use Ruby provided functionality for `.reverse` and `.reverse!`. You may use `.length`.
```
Similar comment to one of @kariabancroft's comments on the `reverse_words` repo: It seems likely that we want an extra blank line above this so that the note ends up in its own paragraph.

Hamled reviewed (approved):
Looks good. I would consider moving the empty string and nil tests into the edge cases section, while moving the long sentence case into the basic tests section.

shrutivanw committed:
Refactoring test structure

Moved nil and empty string tests to be part of edge cases.
Added a new line before note in readme.md to faciliate the rendering.

shrutivanw commented:
Thank you @Hamled I moved the nil and empty string test cases into edge cases and added a newline so readme.md would render correctly.

I think the long string test case belongs in edge cases so I left it there - as is. We can chat further on that one if you feel strongly about it. :-)
